### PR TITLE
Update localstack and aws-cli

### PR DESCRIPTION
**Pull request by wheelsandcogs**
## Why?
SQS seems to keep falling over for me on M1 Mac. Not sure exactly why but we're using quite an old localstack.

## localstack
 - The newer versions of localstack have native arm64 support
 - The default port config now seems to be 4566 (configurable with `EDGE_PORT`)
 - The default SQS provider is now `moto` but has issues so using previous provider `elasticmq` see notes
 - There is now a `/health` endpoint we could ping to see if services have started (e.g. SQS) https://github.com/localstack/localstack#service-health-checks

### Notes
The default port has changed - we could config it back to the same port as before (4576) but all the docs use 4566 and some tooling expects 4566 so in the long run I think it's probably just easier to update our `.env` with the new port.

localstack sqs now includes the account id in the path, so by default you end up with a URL like http://localhost:4566/000000000000/asl-dev

You can configure the value with the env var `TEST_AWS_ACCOUNT_ID` and set it to `queue` for a url like http://localhost:4566/queue/asl-dev when using the newer `moto` provider, but the previous provider `elasticmq` seems to ignore it.

However, the `moto` provider seems to return invalid uuids as the `MessageId` and our app throws validation errors if a `MessageId` is not uuid v4, so we can't use it without modifying our uuid validation. Amazon don't seem to actually guarantee that MessageIds will be UUID v4 so this may be something we need to look at:
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-queue-message-identifiers.html#message-id

## aws-cli
 - Switch to official amazon/aws-cli image which has native arm64 support
 - mesosphere version hasn't been updated in about 4 years
 - Only supports AWS CLI v2 but seems to be a drop-in replacement for our usage

## TLDR;
 - You will need to update your `.env` file with `SQS_URL=http://localhost:4566/000000000000/asl-dev`

### CODE REVIEW

File: conductor.json [hunk 1]
```diff
@@ -323,20 +323,24 @@
...
-        "DEFAULT_REGION": "{{env.SQS_REGION}}"
+        "SQS_PROVIDER": "elasticmq"
       }
     },
     {
       "name": "aws-cli",
-      "image": "mesosphere/aws-cli",
+      "image": "amazon/aws-cli:2.4.6",
+      "depends_on": [
+        "sqs"
```
Comment (wheelsandcogs) [inline conductor.json:337]: Don't run until sqs is at least started (although it doesn't wait until it's running healthily annoyingly).

File: conductor.json [hunk 2]
```diff
@@ -323,20 +323,24 @@
...
+      ],
       "links": [
         "sqs"
       ],
-      "run": "--endpoint-url=http://sqs:4576 sqs create-queue --queue-name=asl-dev --region=eu-west-1 --no-sign-request --no-verify-ssl"
+      "run": "--endpoint-url=http://sqs:4566 sqs create-queue --queue-name=asl-dev --region=eu-west-1 --no-sign-request --no-verify-ssl",
+      "restart": "on-failure"
```
Comment (wheelsandcogs) [inline conductor.json:343]: keep retrying until successful.
